From: Stefan Behrens <sbehrens@giantdisaster.de>
Date: Wed, 23 May 2012 17:57:49 +0200
Patch-mainline: 3.5
Git-commit: 48235a68a3d1db579fc20d9915815228a1825757
Subject: [PATCH] Btrfs: fix false positive in check-integrity on
 unmount

During unmount, it could happen that the integrity checker printed a
warning message "attempt to free ... on umount which is not yet iodone"
which turned out to be a false positive.

Signed-off-by: Stefan Behrens <sbehrens@giantdisaster.de>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/check-integrity.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/btrfs/check-integrity.c
+++ b/fs/btrfs/check-integrity.c
@@ -3338,7 +3338,7 @@ void btrfsic_unmount(struct btrfs_root *
 				btrfsic_block_link_free(l);
 		}
 
-		if (b_all->is_iodone)
+		if (b_all->is_iodone || b_all->never_written)
 			btrfsic_block_free(b_all);
 		else
 			printk(KERN_INFO "btrfs: attempt to free %c-block"
